packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -206,6 +207,26 @@ if (__DEV__) {
     testElement.innerHTML = html;
     return testElement.innerHTML;
   };
+
+  formatTagDEV = function(node: Element): string {
+    let str = '<' + node.nodeName.toLowerCase();
+    const attributeNames = node.getAttributeNames();
+    for (let i = 0; i < attributeNames.length; i++) {
+      if (i > 30) {
+        str += ' ...';
+        break;
+      }
+      const attributeName = attributeNames[i];
+      const value = node.getAttribute(attributeName);
+      let trimmedValue = value;
+      if (value.length > 30) {
+        trimmedValue = value.substr(0, 30) + '...';
+      }
+      str += ' ' + attributeName + '="' + trimmedValue + '"';
```

Comment:
This doesn't escape trimmedValue but I think that's fine. If we wanted to escape we'd need to move the escaping utility from the server folder or copypaste here.